rust/private/rust.bzl

```diff
@@ -866,10 +867,31 @@ rust_shared_library = rule(
         """),
 )
 
+def _proc_macro_dep_transition_impl(_settings, _attr):
+    return {"//:is_proc_macro_dep": True}
+
+_proc_macro_dep_transition = transition(
+    inputs = [],
+    outputs = ["//:is_proc_macro_dep"],
+    implementation = _proc_macro_dep_transition_impl,
+)
+
 rust_proc_macro = rule(
     implementation = _rust_proc_macro_impl,
     provides = _common_providers,
-    attrs = dict(_common_attrs.items()),
+    attrs = dict(
+        _common_attrs.items(),
+        _allowlist_function_transition = attr.label(default = Label("//tools/allowlists/function_transition_allowlist")),
```

Comment:
curious how this works, but is good to know about

Reply:
This is doing a couple of things and deserved a comment; added a comment about it.